tests/sentry/backup/test_correctness.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+
+import io
+import os
+from pathlib import Path
+from typing import NewType, Protocol
+
+from dateutil import parser
+from jsondiff import diff
+from pydantic import FilePath, PositiveInt
+
+from sentry.db.postgres.roles import in_test_psql_role_override
+from sentry.runner.commands.backup import exporter, importer
+from sentry.utils import json
+from sentry.utils.json import JSONData
+from sentry.utils.pytest.fixtures import django_db_all
+
+ComparatorName = NewType("ComparatorName", str)
+ModelName = NewType("ModelName", str)
+
+
+class InstanceID:
+    """Every entry in the generated backup JSON file should have a unique model+pk combination, which serves as its identifier."""
+
+    def __init__(self, model: ModelName, pk: PositiveInt):
+        self.model = model
+        self.pk = pk
+
+    def __eq__(self, other):
+        return (
+            hasattr(other, "model")
```

Comment:
Is this only comparing the `model` and `pk` fields? Do we care about `fields` too?

Reply:
We use the combination of `model` + `pk` to generate a unique id for each entry. We then compare these entries' contents (ie, the `fields`) to perform the actual diffing. The `InstanceID`, the class the `__eq__` method is defined on, only represents this unique ID, not the entire object it identifies.

Reply:
I've added a `TODO` to think a bit more about how this works in practice, but I think its fine for these first few "get the test suite started" commits.